core/internal/cltest/cltest.go

```diff
@@ -1331,51 +1331,25 @@ func BatchElemMustMatchParams(t *testing.T, req rpc.BatchElem, hash common.Hash,
 	}
 }
 
-type SimulateIncomingHeadsArgs struct {
-	StartBlock, EndBlock int64
-	HeadTrackables       []httypes.HeadTrackable
-	Blocks               *Blocks
-}
-
 // SimulateIncomingHeads spawns a goroutine which sends a stream of heads and closes the returned channel when finished.
-func SimulateIncomingHeads(t *testing.T, args SimulateIncomingHeadsArgs) (done chan struct{}) {
-	t.Helper()
-	lggr := logger.TestLogger(t).Named("SimulateIncomingHeads")
-	lggr.Infof("Simulating incoming heads from %v to %v...", args.StartBlock, args.EndBlock)
-
-	if args.EndBlock > args.StartBlock {
-		if l := 1 + args.EndBlock - args.StartBlock; l > int64(len(args.Blocks.Heads)) {
-			t.Fatalf("invalid configuration: too few blocks %d for range length %d", len(args.Blocks.Heads), l)
-		}
-	}
-
+func SimulateIncomingHeads(t *testing.T, heads []*evmtypes.Head, headTrackables ...httypes.HeadTrackable) (done chan struct{}) {
```

Comment:
I was having a really hard time understanding this when debugging so I tried to simplify the API.